lib/friendly_id/slugged.rb

```diff
@@ -300,7 +302,14 @@ def should_generate_new_friendly_id?
     end
 
     def resolve_friendly_id_conflict(candidates)
-      [candidates.first, SecureRandom.uuid].compact.join(friendly_id_config.sequence_separator)
+      candidate = candidates.first
+      uid = SecureRandom.uuid
+      if candidate && friendly_id_config.slug_limit
+        max_candidate_size = friendly_id_config.slug_limit - uid.size - friendly_id_config.sequence_separator.size
+        max_candidate_size = [max_candidate_size, 0].max
+        candidate = candidate[0...max_candidate_size]
+      end
+      [candidate, uid].compact.join(friendly_id_config.sequence_separator)
```

Comment:
OK just one last comment. I would use the 'extract method' refactor here:

```ruby
def resolve_friendly_id_conflict(candidates)
  uuid = SecureRandom.uuid
  [
    apply_slug_limit(candidates.first, uuid), 
    uuid
  ].compact.join(friendly_id_config.sequence_separator)
end

def apply_slug_limit(candidate, uuid)
  return candidate unless candidate && friendly_id_config.slug_limit
  
  candidate[0...slug_limit(uuid)]
end
private :apply_slug_limit

def slug_limit(uuid)
  [
    friendly_id_config.slug_limit - uuid.size - friendly_id_config.sequence_separator.size,
    0
  ].max
end
private :max_candidate_size_for_uuid
```

I haven't tried the code, but would something along those lines work?
My rationale is that it just makes the original method easier to reason about rather than having to read the code a few times to figure out what's going on with the conditional.  It also means others can use this functionality without rewriting it if they override `resolve_friendly_id_conflict`. 😄